darts/tests/models/forecasting/test_torch_forecasting_model.py

```diff
@@ -216,6 +223,337 @@ def test_manual_save_and_load(self):
+            for all but one test.
+            Note: Using DLinear since it supports both past and future covariates
+            """
+
+            def create_DLinearModel(
+                model_name: str,
+                save_checkpoints: bool = False,
+                likelihood: Likelihood = None,
+            ):
+                return DLinearModel(
+                    input_chunk_length=4,
+                    output_chunk_length=1,
+                    kernel_size=5,
+                    model_name=model_name,
+                    work_dir=self.temp_work_dir,
+                    save_checkpoints=save_checkpoints,
+                    likelihood=likelihood,
+                    random_state=42,
+                    force_reset=True,
+                )
+
+            model_dir = os.path.join(self.temp_work_dir)
+            manual_name = "save_manual"
+            auto_name = "save_auto"
+            # create manually saved model checkpoints folder
+            checkpoint_path_manual = os.path.join(model_dir, manual_name)
+            os.mkdir(checkpoint_path_manual)
+            checkpoint_file_name = "checkpoint_0.pth.tar"
+            model_path_manual = os.path.join(
+                checkpoint_path_manual, checkpoint_file_name
+            )
+
+            model_auto_save = create_DLinearModel(
+                auto_name,
+                save_checkpoints=True,
+                likelihood=GaussianLikelihood(prior_mu=0.5),
+            )
+            model_auto_save.fit(self.series, epochs=1)
+            pred_auto = model_auto_save.predict(n=4, series=self.series)
+
+            model_manual_save = create_DLinearModel(
+                manual_name,
+                save_checkpoints=False,
+                likelihood=GaussianLikelihood(prior_mu=0.5),
+            )
+            model_manual_save.fit(self.series, epochs=1)
+            model_manual_save.save(model_path_manual)
+            pred_manual = model_manual_save.predict(n=4, series=self.series)
+
+            # predictions are identical when using the same likelihood
+            self.assertTrue(np.array_equal(pred_auto.values(), pred_manual.values()))
+
+            # model with identical likelihood
+            model_same_likelihood = create_DLinearModel(
+                "same_likelihood", likelihood=GaussianLikelihood(prior_mu=0.5)
+            )
+            model_same_likelihood.load_weights(model_path_manual, map_location="cpu")
+            model_same_likelihood.to_cpu()
+            model_same_likelihood.predict(n=4, series=self.series)
+            # cannot check predictions since this model is not fitted, random state is different
+
+            # model with no likelihood
+            model_no_likelihood = create_DLinearModel("no_likelihood", likelihood=None)
+            # state_dict does not have the proper dimensions, expection raised by torch.load
+            with self.assertRaises(RuntimeError):
```

Comment:
Shouldn't this raise an Error from Darts side? likelihood=None != likelihood=GaussianLikelihood() (during the parameter checks)

Reply:
Yup, the check in `load_weights_from_checkpoint()` was skipping the `None` attributes. Solved it!